### DISCUSSION TITLE
Debugging Non-User Defined Code

### DISCUSSION THREAD

**Discussion by ghost** [microsoft/debugpy#1601] [Q&A]
## Environment data

- debugpy version: 1.8.1
- OS: Windows 10
- Python version: 3.10.11
- VS Code

## Actual behavior

I'm experiencing difficulties when trying to debug non-user-defined code (e.g., imported libraries or modules) in a Jupyter notebook using Visual Studio Code (VS Code). While the debugger stops at breakpoints, I am unable to inspect the variables effectively. The cursor stops at breakpoints but remains green instead of yellow, indicating that the debugging session is not fully interactive (the same problem happen even if I use a .py script instead of a jupyter notebook file).
The cursor stops at breakpoints but remains green. I am unable to acquire any information about the variables in the non-user-defined code.

## Expected behavior

When the debugger stops at a breakpoint, the cursor should turn yellow, and it should be possible to inspect the variables used in the .py file

## Steps to reproduce:

Here is my launch.json configuration:
`{`
`"version": "0.2.0",`
`"{`
`  configurations": [`
`    {`
`      "name": "Debugpy",`
`      "type": "debugpy",`
`      "request": "launch",`
`      "program": "${file}",`
`      "console": "integratedTerminal",`
`      "justMyCode": false`
`    }`
`  ]`
`}`
I'm adding some pictures to better show the error. I've added a breakpoint in the package source code (2nd image) and started the debugger. As you can see, the execution stops at the correct position, but the yellow cursor is not stepping into the package code and stays "outside" in my code.
![1](https://github.com/microsoft/debugpy/assets/172576011/5ffb7f24-cb3a-43c8-ac51-ab8eaa9362c5)
![2](https://github.com/microsoft/debugpy/assets/172576011/126a4e16-3fb9-42c7-8955-690ebf05c4dd)

**Comment by ghost** [comment]
I have found the problem, VSCode 1.90 have a sort of bug that cannot allow python module debug!

**Reply by rchiodo** [reply]
Yes it should be this VS code issue here:
https://github.com/microsoft/vscode/issues/214433
